client/attest.go

```diff
@@ -36,6 +37,11 @@ type AttestOpts struct {
 	// Key.Attest() will construct the certificate chain by making GET requests to
 	// the contents of Key.cert.IssuingCertificateURL using this client.
 	CertChainFetcher *http.Client
+	// SevSnpDevice implements the go-sev-guest client.Device interface for
+	// collecting an SEV-SNP attestation, and is expected to be open. Attest
+	// will not close this device. If nil, then attestation will use
```

Comment:
I wonder if this name and even interface should be so specific. We only use one CC hardware device at a time, so long term we could have a generic interface that every CC device type implements. Then each one could provide technology attestation-specific inputs e.g., technology type, different sized nonces based on requirements/best practices, different event log formats, etc.

Reply:
Done.